annotate.el

```diff
@@ -950,18 +968,65 @@ essentially what you get from:
       (delete-overlay ov))
     (set-buffer-modified-p modified-p)))
 
-(defun annotate-create-annotation (start end &optional text)
+(defun annotate-string-empty-p (a)
+  "Is the arg an empty string or null?"
+  (or (null a)
+      (string= "" a)))
+
+(defun annotate-create-annotation (start end &optional text sample)
   "Create a new annotation for selected region."
-  (let ((annotation (or text (read-from-minibuffer "Annotation: "))))
-    (when (not (or (eq nil annotation) (string= "" annotation)))
-      (let ((highlight (make-overlay start end)))
-        (overlay-put highlight 'face 'annotate-highlight)
-        (overlay-put highlight 'annotation annotation))
-      (when (use-region-p)
-        (deactivate-mark))))
-  (save-excursion
-    (goto-char end)
-    (font-lock-fontify-block 1)))
+  (cl-labels ((create-annotation (start end annotation-text)
+                                 (let ((highlight (make-overlay start end)))
+                                   (overlay-put highlight 'face 'annotate-highlight)
+                                   (overlay-put highlight 'annotation annotation)))
+              (move-lines        (start line-count)
+                                 (save-excursion
+                                   (goto-char start)
+                                   (forward-line line-count)
+                                   (beginning-of-line)
+                                   (point)))
+              (go-backward       (start)
```

Comment:
`beginning-of-previous-line`? But why two lines?

Reply:
Oh, I see. It's two lines because that's the "search region" where a moved annotation can be, right? It might be useful to make that a customizable variable.

Reply:
Good idea!

Reply:
> (I'd have appreciated a more descriptive pull request text ;-) )

Sorry!